services/ownerService.js

```diff
@@ -51,7 +51,9 @@ exports.updateHotel = async (id, data) => {
   return hotel
 }
 
-exports.deleteHotel = async (id, isForceDelete) => {
+exports.deleteHotel = async (ownerId, id, isForceDelete) => {
+  const hotel = await Hotel.findById(id)
+  if (hotel.ownerId !== ownerId) throw new ApiError(403, 'Forbidden')
```

Comment:
First check if hotel exists. When it is undefined we will get an error: ownerId is not a Property of undefined.. Something  like that